File: backend/app/schemas/chart.py
from pydantic import BaseModel


class BarChart(BaseModel):
    title: str
    labels: list[str]
    values: list[float]


class PieChart(BaseModel):
    title: str
    labels: list[str]
    values: list[float]


class ChartResponse(BaseModel):
    bar_chart: BarChart | None = None
    pie_chart: PieChart | None = None

File: frontend/types/chart.ts
export interface BarChart {
  title: string;
  labels: string[];
  values: number[];
}

export interface PieChart {
  title: string;
  labels: string[];
  values: number[];
}

export interface ChartResponse {
  bar_chart: BarChart | null;
  pie_chart: PieChart | null;
}